Validation/HLTrigger/interface/HLTGenValObject.h

```diff
@@ -0,0 +1,54 @@
+#ifndef Validation_HLTrigger_HLTGenValObject_h
+#define Validation_HLTrigger_HLTGenValObject_h
+
+//********************************************************************************
+//
+// Description:
+//   This class is a "wrapper" for the Generator level validation code, which unifies various object types.
+//   It handles the different type of objects the code needs to run on: GenParticles, GenJets and event-level energy sums
+//
+// Author : Finn Labe, UHH, Nov. 2021
+//
+//********************************************************************************
+
+
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/JetReco/interface/GenJet.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include <vector>
+
+class HLTGenValObject {
+public:
+
+  // empty constructor
+  HLTGenValObject() {}
+
+  // constructor from GenParticle
+  HLTGenValObject(const reco::GenParticle &p)
+      : p4Polar_(p.p4()), p4Cartesian_(p.p4()) {}
+
+  // constructor from GenJet
+  HLTGenValObject(const reco::GenJet &p)
+      : p4Polar_(p.p4()), p4Cartesian_(p.p4()) {}
+
+  // constructor from LorentzVector (for energy sums)
+  HLTGenValObject(const reco::Candidate::PolarLorentzVector& p)
+      : p4Polar_(p), p4Cartesian_(p) {}
+
+  // destructor
+  ~HLTGenValObject() {}
```

Comment:
probably from my examples but its best to remove this. It does also technically break CMS coding rules about defining a desstructor but not a copy/assignment operator